File: euclidean/c/euclidean.c
#include <stdlib.h>
#include <stdio.h>

int euclidean(int a, int b){
  int temp;
  while(b!=0){
    temp = a%b;
    a = b;
    b = temp;
  }
  return a;
}

int main(int argc, char *argv[]) {
  printf("%d", euclidean(atoi(argv[1]), atoi(argv[2])));
}

File: euclidean/java/Euclidean.java
public class Euclidean{
  public static void main(String[] args){
    System.out.println(euclidean(Integer.parseInt(args[0]), Integer.parseInt(args[1])));
  }

  public static int euclidean(int a, int b){
    int temp = 0;
    while(b != 0){
      temp = a % b;
      a = b;
      b = temp;
    }
    return a;
  }
}
